### ISSUE TITLE
`HTTPError`s have unspecified response body

### ISSUE THREAD

**Issue by kirillgroshkov** [sindresorhus/got#1162]
#### Describe the bug

 - Node.js version: 12.16.2
 - OS & version: MacOS 10.15

#### Actual behavior

I had a BeforeErrorHook that in got@10 had `err.response` defined, so I used `err.response.body` to display/format error. In got@11 the same code tells me that `err.response.body` is undefined.

Briefly:
```typescript
export function gotErrorHook(): BeforeErrorHook {
  return err => {
    console.log(err instanceof HTTPError, err.response.body)
//=> true, undefined
```

#### Expected behavior

`err.response.body` to be defined

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
Can you set up a full example of the bug?

**Comment by szmarczak** [comment]
We have many tests that prove it's defined, for example: https://github.com/sindresorhus/got/blob/1f6ac4597b797e6fe760a7dc11a3db8bf298aa94/test/error.ts#L20

but since Got has been fully rewrited to utilize the native streams new bugs may occur.

**Comment by kirillgroshkov** [comment]
> Can you set up a full example of the bug?

Yep. But first - am I supposed to have `err.response` defined? Like, this API didn't suppose to change or anything? I noticed the change from `GotError` to `RequestError`, but both have `.response` attached.

**Comment by kirillgroshkov** [comment]
> We have many tests that prove it's defined, for example:
> 
> https://github.com/sindresorhus/got/blob/1f6ac4597b797e6fe760a7dc11a3db8bf298aa94/test/error.ts#L20
> 
> but since Got has been fully rewrited to utilize the native streams new bugs may occur.

I see. I'll try to hack together a minimal repro.. Hold on..

**Comment by szmarczak** [comment]
> But first - am I supposed to have err.response defined?

Yep. All `RequestError`s should have a `.response` defined (if the underlying Node.js request emitted it) except for some invalid usage errors.

**Comment by szmarczak** [comment]
The `HTTPError` should always have a `.response` property.

**Comment by kirillgroshkov** [comment]
Ok, classic thing, minimal repro gives me `.response` as defined. I probably did something wrong in my first try. But, the question propagates to `err.response.body`, which is undefined now and it wasn't if I run the same code against got@10. Here's the got@11 minimal repro:

https://runkit.com/kirillgroshkov/5e9dbf8b5ca48500131a30b8

It logs `err.response.body` as undefined

**Comment by kirillgroshkov** [comment]
Same code with `got@10`: https://runkit.com/kirillgroshkov/5e9dc3019747cc001a89d84c

**Comment by szmarczak** [comment]
Ok, I know how to fix this, give me 15 mins.

**Comment by szmarczak** [comment]
I've encountered another problem. I passed only one `beforeError` hook, and there are two. Weird...

**Comment by szmarczak** [comment]
Fixed in d914a7e7a0d97b03745b9b47623e4ac3532b4dae

**Comment by szmarczak** [comment]
Released `11.0.1`, thanks for reporting!
